extensions/data-plane/data-plane-http/src/main/java/org/eclipse/dataspaceconnector/dataplane/http/DataPlaneHttpExtension.java

```diff
@@ -54,11 +61,21 @@ public String name() {
     @Override
     public void initialize(ServiceExtensionContext context) {
         var monitor = context.getMonitor();
+        var sinkPartitionSize = getSinkPartitionSize(context);
 
         @SuppressWarnings("unchecked") var sourceFactory = new HttpDataSourceFactory(httpClient, retryPolicy, monitor, vault);
         pipelineService.registerFactory(sourceFactory);
 
-        var sinkFactory = new HttpDataSinkFactory(httpClient, executorContainer.getExecutorService(), 5, monitor);
+        var sinkFactory = new HttpDataSinkFactory(httpClient, executorContainer.getExecutorService(), sinkPartitionSize, monitor);
         pipelineService.registerFactory(sinkFactory);
     }
+
+    @NotNull
+    private Integer getSinkPartitionSize(ServiceExtensionContext context) {
+        try {
+            return context.getSetting(EDC_DATAPLANE_HTTP_SINK_PARTITION_SIZE, DEFAULT_PART_SIZE);
+        } catch (NumberFormatException e) {
```

Comment:
`NumberFormatException` is already handled by the `Config` service implementation, so it's not needed to be caught